packages/ciphernode/evm/src/ciphernode_registry_sol.rs

```diff
@@ -0,0 +1,162 @@
+use actix::{Actor, Addr, AsyncContext, Recipient, WrapFuture};
+use alloy::{
+    eips::BlockNumberOrTag,
+    primitives::{Address, LogData, B256},
+    rpc::types::Filter,
+    sol,
+    sol_types::SolEvent,
+};
+use anyhow::Result;
+use enclave_core::{EnclaveEvent, EventBus};
+
+use crate::helpers::{self, create_readonly_provider, ReadonlyProvider};
+
+sol! {
```

Comment:
Is there a way to inherit this directly from the Enclave repo so we don't duplicate code and can maintain a single source of truth?

Reply:
I did look into it but have not really found a solution that looked workable. Really happy to hear suggestions.

Reply:
Actually I think I have found a possibility will test